tsdb/head.go

```diff
@@ -466,11 +471,16 @@ func (h *Head) loadWAL(r *wal.Reader, multiRef map[uint64]uint64) (err error) {
 		}
 	}()
 
+Outer:
 	for d := range decoded {
 		switch v := d.(type) {
 		case []record.RefSeries:
 			for _, s := range v {
-				series, created := h.getOrCreateWithID(s.Ref, s.Labels.Hash(), s.Labels)
+				series, created, err := h.getOrCreateWithID(s.Ref, s.Labels.Hash(), s.Labels)
+				if err != nil {
+					errCh <- err
```

Comment:
@brian-brazil the channel size was set to 2 because we could add error to channel even here in addition to the decoding channel. Probably this is not a good design to do so and a variable should be enough. Also draining the `decoded` channel upon this error is missing, will add that.

Reply:
Without looking at the code, I'd expect there to be 2 channels in that case.

Reply:
And with a quick look, no need for channels. Just 2 variables are enough :)